test/extended/cpu_partitioning/crio.go

```diff
@@ -276,63 +275,68 @@ func collectContainerInfo(ctx context.Context, cfg *rest.Config, c kubernetes.In
 	}
 
 	pod := &pods.Items[0]
+	podName := pod.Name
+	podNamespace := pod.Namespace
+	containerName := pod.Spec.Containers[0].Name
 
-	var out []byte
 	getInfo := []string{
 		"chroot",
 		"/rootfs",
 		"/bin/bash",
 		"-c", collectionScript}
 
-	if err := wait.PollWithContext(ctx, 15*time.Second, time.Minute, func(context.Context) (done bool, err error) {
-		out, err = execCommandOnPod(cfg, c, pod, getInfo)
-		if err != nil {
-			return false, err
-		}
+	backoff := wait.Backoff{
+		Duration: 2 * time.Second,
+		Steps:    5,
+		Factor:   2.0,
+		Jitter:   0.1,
+	}
 
-		return len(out) != 0, nil
-	}); err != nil {
-		return nil, err
+	execOptions := e2epod.ExecOptions{
+		Command:            getInfo,
+		PodName:            podName,
+		Namespace:          podNamespace,
+		ContainerName:      containerName,
+		Stdin:              nil,
+		CaptureStdout:      true,
+		CaptureStderr:      true,
+		PreserveWhitespace: false,
 	}
 
 	info := []crioContainerData{}
-	if err := json.Unmarshal(out, &info); err != nil {
-		return nil, fmt.Errorf("error parsing container output to json: %w", err)
-	}
+	err = retry.OnError(backoff, shouldRetryExec,
+		func() error {
+			out, _, execErr := e2epod.ExecWithOptions(oc.KubeFramework(), execOptions)
```

Comment:
the retry logic seems nice, but why using this code is better than the previous code? just avoiding code duplication or other reasons?

Reply:
yeah, noticed this had some nicer options (like logging the request) and figured we could avoid code duplications and hopefully less things for us to manage and worry about